.github/workflows/pattern-metrics.yaml

```
@@ -0,0 +1,120 @@
+name: Monthly issue metrics
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '3 2 1 * *'
+
+permissions:
+  issues: write
+  pull-requests: read
+
+jobs:
+  issue-metrics:
+    name: issue metrics
+    runs-on: ubuntu-latest
+
+    steps:
+
+    - name: Get dates for last month
+      shell: bash
+      run: |
+        # Calculate the first day of the previous month
+        first_day=$(date -d "last month" +%Y-%m-01)
+
+        # Calculate the last day of the previous month
+        last_day=$(date -d "$first_day +1 month -1 day" +%Y-%m-%d)
+
+        #Set an environment variable with the date range
+        echo "$first_day..$last_day"
+        echo "last_month=$first_day..$last_day" >> "$GITHUB_ENV"
+
+    - name: Check out the code so we can get the CODEOWNERS names
+      uses: actions/checkout@v4
```

```suggestion
      uses: actions/checkout@v4
        with:
          sparse-checkout: |
            .github/CODEOWNERS
          sparse-checkout-cone-mode: false
```

I guess it is fine, in terms of performance, to checkout the whole repo?

We could only checkout the `.github/CODEOWNERS` file, if that is the only think we need? Of course then we would have to change that again, if we ever want to use any other files from the repo.

Code above based on [this](https://github.com/actions/checkout#fetch-only-a-single-file).

---

That is a good idea!

---

Fun fact 1: This created a 4 second performance improvement :)

Fun fact 2: I had messed up the indentation int he YAML, which made the GHA fail. Was a quick fix though.